### PR TITLE
contrib/jinzhu/gorm: Add custom tags.

### PR DESCRIPTION
**Pull request by cube2222**
Implements #716

### CODE REVIEW

File: contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go [hunk 1]
```diff
@@ -303,3 +303,44 @@ func TestContext(t *testing.T) {
...
+			return scope.SQLVars[3]
+		}),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer db.Close()
+	db.AutoMigrate(&Product{})
+
+	t.Run("create", func(t *testing.T) {
```
Comment (gbbr) [inline contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go:324]: I don't think it makes sense to have a sub-test when there is only one of them.

File: contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go [hunk 2]
```diff
@@ -303,3 +303,44 @@ func TestContext(t *testing.T) {
...
+
+		db = WithContext(ctx, db)
+		db.Create(&Product{Code: "L1212", Price: 1000})
+
+		parentSpan.Finish()
+
+		spans := mt.FinishedSpans()
+		assert.True(len(spans) >= 3)
+
+		span := spans[len(spans)-3]
```
Comment (gbbr) [inline contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go:338]: I think this warrants an explanation. An inline comment. What's 3?

File: contrib/gopkg.in/jinzhu/gorm.v1/option.go [hunk 3]
```diff
@@ -8,13 +8,16 @@ package gorm
 import (
 	"math"
 
+	"gopkg.in/jinzhu/gorm.v1"
```
Comment (gbbr) [inline contrib/gopkg.in/jinzhu/gorm.v1/option.go:11]: This import group goes last.

File: contrib/gopkg.in/jinzhu/gorm.v1/option.go [hunk 4]
```diff
@@ -28,6 +31,7 @@ func defaults(cfg *config) {
 	} else {
 		cfg.analyticsRate = math.NaN()
 	}
+	cfg.tagFns = make(map[string]func(scope *gorm.Scope) interface{})
```
Comment (gbbr) [inline contrib/gopkg.in/jinzhu/gorm.v1/option.go:34]: Let's only make this map if `WithCustomTag` is called at least once.

**Review by gbbr — COMMENTED** [review]
Nice work! Thanks! A few comments...

**Comment by cube2222** [comment]
Thanks for the review! Please take a look now.

File: contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go [hunk 5]
```diff
@@ -303,3 +303,44 @@ func TestContext(t *testing.T) {
...
+
+	db = WithContext(ctx, db)
+	db.Create(&Product{Code: "L1212", Price: 1000})
+
+	parentSpan.Finish()
+
+	spans := mt.FinishedSpans()
+	assert.True(len(spans) >= 3)
+
+	// We deterministically expect the span to be the third last,
```
Comment (gbbr) [inline contrib/gopkg.in/jinzhu/gorm.v1/gorm_test.go:337]: 👍

**Review by gbbr — APPROVED** [review]
Looks great! Thanks!

**Pull request merged**